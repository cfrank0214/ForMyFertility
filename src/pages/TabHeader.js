import React from 'react';
import { connect } from 'react-redux';
import {
  Button, Left, Body, Right, Title, Icon, Header
} from 'native-base';
import {
  actions,
} from 'react-redux-form/native';
import {
  DrawerActions, NavigationActions
} from 'react-navigation';

export class TabHeader extends React.PureComponent {

  handleBackButton({ questionNum, dispatch, navigation }) {
    dispatch(actions.change('fertilityQuestions.questionNum', --questionNum));
    navigation.dispatch(NavigationActions.back())
  }
  render() {
    let questionNum = this.props.fertilityQuestions ? this.props.fertilityQuestions.questionNum : 1;
    const { dispatch } = this.props;
    const navigation = this.props.navigation;
    return (
      <Header>
        <Left style={{ flex: 1 }}>
          <Button
            transparent
            onPress={() => this.handleBackButton({ questionNum, dispatch, navigation })}
          >
            <Icon name="arrow-back" />
          </Button>
        </Left>


        <Body style={{ flex: 3 }}>
          <Title style={{ alignSelf: "center" }}>{this.props.title || 'For My Fertility'}</Title>
        </Body>
        <Right style={{ flex: 1 }}>
          <Button
            transparent
            onPress={() => navigation.dispatch(DrawerActions.openDrawer())}
          >
            <Icon name="menu" />
          </Button>
        </Right>
      </Header>
    );
  }
}

function mapStateToProps(state) {
  return {
    nav: state.nav,
    fertilityQuestions: state.fertilityQuestions
  };
}

export default connect(mapStateToProps)(TabHeader);