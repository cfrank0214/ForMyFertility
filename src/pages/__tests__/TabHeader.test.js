/**
 * @jest-environment jsdom
 */
 import React from 'react'
 import { shallow } from 'enzyme';
 import renderer from 'react-test-renderer';
 import { TabHeader } from '../TabHeader';
 import {
  Button, Left, Body, Right, Header
} from 'native-base';
 
 describe('>>>TabHeader --- Shallow Render REACT COMPONENTS',()=>{
   let wrapper
   
   beforeEach(()=>{

    const props = {
      fertilityQuestions: jest.fn(),
      navigation: jest.fn(),
      dispatch: jest.fn()
    }
       wrapper = shallow(<TabHeader {...props}  />)
       
   })
 
   it('renders correctly', () => {
     const tree = renderer.create(
       <TabHeader />
       ).toJSON();
     expect(tree).toMatchSnapshot();
   });
 
   it('+++ render the DUMB component', () => {
      expect(wrapper.length).toEqual(1)
   });
   it('contains 2 <Button /> components', () => {
     expect(wrapper.find(Button).length).toBe(2);
   });
   it('contains 1 <Left /> component', () => {
     expect(wrapper.find(Left).length).toBe(1);
   });
   it('contains 1 <Body /> component', () => {
    expect(wrapper.find(Body).length).toBe(1);
  });
  it('contains 1 <Right /> component', () => {
    expect(wrapper.find(Right).length).toBe(1);
  });
  it('contains 1 <Header /> component', () => {
    expect(wrapper.find(Header).length).toBe(1);
  });
 });