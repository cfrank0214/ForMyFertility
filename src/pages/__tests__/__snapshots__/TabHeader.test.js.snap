// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>TabHeader --- Shallow Render REACT COMPONENTS renders correctly 1`] = `
<View
  onLayout={[Function]}
>
  <View
    style={
      Object {
        "backgroundColor": "#F8F8F8",
        "borderBottomColor": "#a7a6ab",
        "borderBottomWidth": 0.5,
        "elevation": 3,
        "flexDirection": "row",
        "height": 64,
        "justifyContent": "center",
        "left": 0,
        "paddingLeft": 6,
        "paddingRight": 10,
        "paddingTop": 18,
        "right": 0,
        "shadowColor": undefined,
        "shadowOffset": undefined,
        "shadowOpacity": undefined,
        "shadowRadius": undefined,
        "top": 0,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "flex-start",
            "alignSelf": "center",
            "flex": 1,
          },
          Object {
            "flex": 1,
          },
        ]
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": null,
            "backgroundColor": "transparent",
            "borderBottomWidth": null,
            "borderColor": null,
            "borderLeftWidth": null,
            "borderRadius": 5,
            "borderRightWidth": null,
            "borderTopWidth": null,
            "borderWidth": undefined,
            "elevation": 0,
            "flexDirection": "row",
            "height": 45,
            "justifyContent": "space-between",
            "marginLeft": -3,
            "opacity": 1,
            "paddingBottom": 6,
            "paddingLeft": 4,
            "paddingRight": 6,
            "paddingTop": 6,
            "shadowColor": null,
            "shadowOffset": null,
            "shadowOpacity": null,
            "shadowRadius": null,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 12,
              },
              Array [
                Object {
                  "color": "#000",
                  "fontSize": 30,
                },
                Object {
                  "color": "#007aff",
                  "fontSize": 34,
                  "marginLeft": 1,
                  "marginRight": 2,
                  "marginTop": 0,
                  "paddingTop": 1,
                },
              ],
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "flex": 1,
          },
          Object {
            "flex": 3,
          },
        ]
      }
    >
      <Text
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 17,
              "fontWeight": "700",
              "marginLeft": undefined,
              "paddingLeft": 4,
              "paddingTop": 1,
              "textAlign": "center",
            },
            Object {
              "alignSelf": "center",
            },
          ]
        }
      >
        For My Fertility
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "flex-end",
            "alignSelf": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "flex-end",
          },
          Object {
            "flex": 1,
          },
        ]
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": null,
            "backgroundColor": "transparent",
            "borderBottomWidth": null,
            "borderColor": null,
            "borderLeftWidth": null,
            "borderRadius": 50,
            "borderRightWidth": null,
            "borderTopWidth": null,
            "borderWidth": undefined,
            "elevation": 0,
            "flexDirection": "row",
            "height": 45,
            "justifyContent": "space-between",
            "marginRight": -9,
            "opacity": 1,
            "paddingBottom": 6,
            "paddingLeft": 15,
            "paddingRight": 15,
            "paddingTop": 6,
            "shadowColor": null,
            "shadowOffset": null,
            "shadowOpacity": null,
            "shadowRadius": null,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 12,
              },
              Array [
                Object {
                  "color": "#000",
                  "fontSize": 30,
                },
                Object {
                  "color": "#007aff",
                  "fontSize": 31,
                  "marginLeft": 2,
                  "marginRight": 0,
                  "marginTop": 0,
                  "paddingTop": 2,
                },
              ],
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
</View>
`;
